File: fvc-api/src/main/java/sep490g65/fvcapi/entity/TeamMember.java
package sep490g65.fvcapi.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(
        name = "team_members",
        uniqueConstraints = @UniqueConstraint(name = "uk_team_user_unique", columnNames = {"team_id", "user_id"})
)
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class TeamMember extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private LocalDateTime joinedAt;

    private LocalDateTime leftAt;

    @Column(length = 20)
    private String status; // ACTIVE | REMOVED
}

File: fvc-frontend/src/types/teammember.ts
export type TeamMemberStatus = "ACTIVE" | "REMOVED";

export interface TeamMemberDto {
  id: string;
  teamId: string;
  userId: string;
  status: TeamMemberStatus;
  joinedAt?: string;
  leftAt?: string;
}

export interface TeamMemberAddRequest {
  userId: string;
  note?: string;
}

export interface TeamMemberRemoveRequest {
  note?: string;
}

export interface TeamMemberBulkAddRequest {
  userIds: string[];
  note?: string;
}


